File: javascript/p1518.js
/**
 * @param {number} numBottles
 * @param {number} numExchange
 * @return {number}
 */
export var numWaterBottles = function (numBottles, numExchange) {
  let drank = numBottles;

  while (numBottles >= numExchange) {
    drank += Math.trunc(numBottles / numExchange);
    numBottles =
      Math.trunc(numBottles / numExchange) + (numBottles % numExchange);
  }

  return drank;
};

File: python/p1518.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        drank = numBottles
        
        while numBottles >= numExchange:
            drank += numBottles // numExchange
            numBottles = numBottles // numExchange + numBottles % numExchange
            
        return drank